Strengthen well known check-cfg names and values test

Urgau opened the pull request:
https://github.com/rust-lang/rust/pull/118494 is changing the implementation of how we expect well known check-cfg names and values, but we currently don't have a test that checks every well known only some of them.

This PR therefore strengthen our well known names/values test to include all of the configs to at least avoid unintended regressions and validate new entry.

*this PR also contains some drive-by consolidation of unexpected `target_os`, `target_arch` into a single file*

r? @nnethercote (maybe? feel free to re-assign)

Urgau committed:
Strengthen well known check-cfg names and values test

nnethercote commented on compiler/rustc_session/src/config.rs:
```diff
@@ -1422,6 +1422,9 @@ impl CheckCfg {
         // NOTE: This should be kept in sync with `default_configuration`
+        //
+        // When adding a new config here you should also update
+        // `tests/ui/check-cfg/well-known-values.rs`.
```
👍

nnethercote commented on tests/ui/check-cfg/well-known-values.rs:
```diff
@@ -1,41 +1,90 @@
-// This test check that we lint on non well known values and that we don't lint on well known
-// values
+// This test check that we recogniez all the well known config names
```
```suggestion
// This test check that we recognize all the well known config names
```

nnethercote commented on tests/ui/check-cfg/well-known-values.rs:
```diff
@@ -1,41 +1,90 @@
+
+#[cfg(any(
+    doc = "_UNEXPECTED_VALUE",
+    //~^ WARN unexpected `cfg` condition value
```
I would put this list in alphabetical order, and maybe even add `tidy-alphabetical-{start,end}` markers so it is guaranteed to stay in order.

Urgau replied:
Great, and to make sure it works as expected I've tested it locally with success.

nnethercote commented on tests/ui/check-cfg/well-known-values.rs:
```diff
@@ -1,41 +1,90 @@
-fn target_has_atomic_invalid() {}
-
 #[cfg(target_has_atomic = "8")]
 fn target_has_atomic() {}
```
It's not clear to me why some of these remain, when they are now mentioned in the big list above? But some were removed.

nnethercote replied:
The linux_misspell one makes sense to remain, but I don't understand the rest.

Urgau replied:
We still need to test that the combination of expected well known names and values do not produce a warning, but we do not need to test all of them, so I chose some that seemed relevant (with/without a value, both at the same time). It's very arbitrary and I'm happy to change it if desired.

nnethercote replied:
A comment explaining exactly that would be helpful :)

Urgau replied:
Done. I've improved the comment at the start of the section.

nnethercote commented:
> this PR also contains some drive-by consolidation of unexpected target_os, target_arch into a single file

I generally put these kinds of changes into separate commits. It makes the reviewer's job easier when unrelated changes are separated out.

Urgau committed:
Avoid target_os and target_arch in some check-cfg tests

as they unnecessarily clutter the diagnostic output and make the
experience of adding a new target to the compiler more painful than
it should be.

target_os and target_arch are still being tested in the
well-known-values.rs test, but in one place.

nnethercote commented on tests/ui/check-cfg/well-known-values.rs:
```diff
@@ -1,41 +1,106 @@
+// being noticed we pass them a obviously wrong value so the
+// diagnostic prints the list of expected values.
+//
+// This is in alphabetical order.
```
Don't need this line any more now that the tidy markers are present.

With that, I think it's ready to go.

@bors delegate=Urgau

bors commented:
:v: @Urgau, you can now approve this pull request!

If @nnethercote told you to "`r=me`" after making some further change, please make that change, then do `@bors r=@nnethercote`
<!-- homu: {"type":"Delegated","delegator":"nnethercote","delegate":"Urgau"} -->

Urgau commented:
@bors r=@nnethercote rollup=always

bors commented:
@Urgau: :key: Insufficient privileges: Not in reviewers

Urgau commented:
@bors r=nnethercote

bors commented:
@Urgau: :key: Insufficient privileges: Not in reviewers

GuillaumeGomez commented:
Weird. Oh well.

@bors r=nnethercote rollup

GuillaumeGomez commented:
Failed in https://github.com/rust-lang/rust/pull/118744.

workingjubilee commented:
@bors r-

Urgau commented:
I've rebased and re-blessed the diagnostic output (new win7 target).

@rustbot ready

GuillaumeGomez commented:
@bors r=nnethercote

bors commented:
:pushpin: Commit 4c1671674e97fa8a60f3999a3770f0c32499c584 has been approved by `nnethercote`

It is now in the [queue](https://bors.rust-lang.org/homu/queue/rust) for this repository.

<!-- @bors r=nnethercote 4c1671674e97fa8a60f3999a3770f0c32499c584 -->
<!-- homu: {"type":"Approved","sha":"4c1671674e97fa8a60f3999a3770f0c32499c584","approver":"nnethercote","queue":"https://bors.rust-lang.org/homu/queue/rust"} -->
